third_party/move/move-vm/runtime/src/tracing.rs

```diff
@@ -68,17 +76,13 @@ pub(crate) fn trace(
     interp: &Interpreter,
 ) {
     if *TRACING_ENABLED {
-        let f = &mut *LOGGING_FILE.lock().unwrap();
-        writeln!(
-            f,
-            "{}-{:?},{},{},{:?}",
-            process::id(),
-            thread::current().id(),
-            function_desc.pretty_string(),
-            pc,
-            instr,
-        )
-        .unwrap();
+        let buf_writer = &mut *LOGGING_FILE_WRITER.lock().unwrap();
+        buf_writer
+            .write_fmt(format_args!("{},{}\n", function_desc.pretty_string(), pc,))
+            .unwrap();
+        if *SINGLE_STEP_FLUSHING {
```

Comment:
Looks like this is always enabled if tracing in the move vm is enabled.

If so, there does not seem to be any difference in using a buffered write: we end up flushing after write.

Was there any measurable performance increase after adding buffered-writes-with-frequent-flushes vs. using plain unbuffered writes (of the same amount of data)?

Reply:
It is only enabled in testing

Reply:
Curious: do you have some numbers for the performance gain by enabling write buffering?